README.md

```diff
@@ -1,158 +1,278 @@
-# Java REST wrapper for Microsoft Azure Notification Hubs
+[ ![Download](https://api.bintray.com/packages/microsoftazuremobile/SDK/Notification-Hubs-Java-Backend-SDK/images/download.svg) ](https://bintray.com/microsoftazuremobile/SDK/Notification-Hubs-Java-Backend-SDK/_latestVersion)
 
+# Microsoft Azure Notification Hubs Java SDK
 
-Java Back-end SDK for [Microsoft Azure Notification Hubs](https://azure.microsoft.com/en-us/services/notification-hubs/) which wraps the [REST APIs](https://docs.microsoft.com/en-us/rest/api/notificationhubs/).
+This repository contains source code for the Java SDK for [Microsoft Azure Notification Hubs](https://azure.microsoft.com/en-us/services/notification-hubs/).
 
-Implements most of the Notification Hubs REST operations to perform Hub CRUDs, registration/installation management and send notification.
+### Note on FCM and GCM Support
 
-## What's New
+The Azure Notification Hubs SDK has support for Firebase Cloud Messaging (FCM) via the Legacy HTTP APIs which is compatible with GCM.  For registrations, use the `FirebaseRegistration` to create registrations, as the `GcmRegistration` class has been deprecated.  For more information, read the [Azure Notification Hubs and Google Firebase Cloud Messaging migration](https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-gcm-to-fcm)
 
-* FCM and GCM support
+For installations, use the `NotificationPlatform.Gcm` for all FCM registrations which use the legacy HTTP APIs.  Currently, we do not have full support for FCM and calling `NotificationPlatform.Fcm` will result in an exception.
 
-## Use
+```java
+Installation i = new Installation();
+
+// Uses the FCM Legacy API
+i.setPlatform(NotificationPlatform.Gcm);
+```
 
-### Compile and build
-Use [Maven](http://maven.apache.org/).
+## Building the Azure Notification Hubs Java SDK
 
-To build:
+To build, use [Maven](http://maven.apache.org/):
 
 ```bash
 $ cd NotificationHubs
 $ mvn package
 ```
 
-### Code
+## Getting Started
+
+To get started, you can find all the classes in the `com.windowsazure.messaging` package, for example:
+
+```java
+import com.windowsazure.messaging.NotificationHub;
+```
+
+The Azure Notification Hubs SDK for Java support both synchrnous and asynchrnous operations on `NotificationHub/NotificationHubClient` and `NamespaceManager/NamespaceManagerClient`.  The asynchronous APIs are supported using the a `org.apache.http.concurrent.FutureCallback` interface.
```

Comment:
synchronous and asynchronous are misspelled here

Reply:
> using the a

has an extra word "a".

Reply:
Removed